keps/sig-cloud-provider/0002-cloud-controller-manager.md

```diff
@@ -266,9 +298,317 @@ In case of the cloud-controller-manager, the deployment should be deleted using
+trees. Code which needs to be consumed by other repos (Eg CSI plugins, shared controllers, ...) should not be in the 
+K8s/cloud-provider repo but in an appropriate K8s SIGs/library repo.
+    
+The code which needs to be shared can be broken into several types. 
+
+There is code which properly belongs in the various cloud-provider repos. The classic example of this would be the 
+implementations of the cloud provider interface. (Eg. [gce](https://github.com/kubernetes/kubernetes/tree/master/pkg/cloudprovider/providers/gce) 
+or [aws](https://github.com/kubernetes/kubernetes/tree/master/pkg/cloudprovider/providers/aws)) These sections of code 
+need to be shared until we can remove all cloud provider dependencies from K8s/K8s. (i.e. When KAS, KCM and Kubelet no 
+longer contain a cloud-provider flag and no longer depend on either the cloud provider interface or any cloud provider 
+implementations) At that point they should be permanently moved to the individual provider repos. I would suggest that 
+as long as the code is shared it be in vendor for K8s/cloud-provider. We would want to create a separate Staging repo 
+in K8s/K8s for each cloud-provider. 
+
+The majority of the controller-manager framework and its dependencies also needs to be shared. This is code that would 
+be shared even after cloud providers are removed from K8s/K8s. As such it probably makes sense to make the controller 
+manager framework its own K8s/K8s Staging repo. 
+
+Lastly we have any controllers which might run in either KCM or CCM. This includes things like NodeIpamController, 
+which will be shared both in the short and long term. Currently it needs to take a cloud provider to allow it to do GCE 
+CIDR management. We could handle this by leaving the cloud provider interface with the controller manager framework 
+code. The GCE controller manager could then inject the cloud provider for that controller. For everyone else 
+(especially the KCM) NodeIpamController is interesting because currently everyone needs its generic behavior for things 
+like ranges. However Google then wires in the cloud provider to provide custom functionality on things like CIDRs. The 
+thought then is that in the short term we allow it to be run in either KCM or CCM. Most cloud providers would run it in 
+the KCM, Google would run it in the CCM. When we are ready to move cloud provider code out of K8s/K8s, we remove the 
+cloud provider code from the version which is in K8s/K8s and continue to have flags to control if it runs. 
+K8s/Cloud-Provider-Google could then have an enhanced version which is run in the CCM. Other controllers such as Route 
+and Service needs to run in either the KCM or CCM. For things like K8s/K8s e2e tests we will always want these 
+controllers in the K8s/K8s repo. Having it in the K8s/K8s repo is also useful for keeping the behavior of these sort of 
+core systems consistent.
+
+#### Staging
```

Comment:
I personally think we should pitch moving code to k8s/k8s/staging in a separate KEP. Given we'll want every in-tree cloud provider to approve it might be easier to consume if it's presented in a more consise / smaller scoped KEP.

Reply:
Adding to KEP 13